File: frequency_counter_using_a_hash_map/fcuhm.ts
function countFrequencies<T>(elements: T[]): Map<T, number> {
    const counts = new Map<T, number>();

    for (const item of elements) {
        const currentCount = counts.get(item);

        if (currentCount === undefined) {
            counts.set(item, 1);
        } else {
            counts.set(item, currentCount + 1);
        }
    }

    return counts;
}

const words = ["a", "b", "a", "c", "b", "a"];
const numbers = [10, 20, 10, 30, 20, 10];

console.log(Array.from(countFrequencies(words).entries()));
console.log(Array.from(countFrequencies(numbers).entries()));

File: frequency_counter_using_a_hash_map/fcuhm.py
from collections.abc import Hashable
from typing import TypeVar

T = TypeVar("T", bound=Hashable)


def count_frequencies(elements: list[T]) -> dict[T, int]:
    counts: dict[T, int] = {}

    for item in elements:
        if item in counts:
            counts[item] += 1
        else:
            counts[item] = 1

    return counts


words = ["a", "b", "a", "c", "b", "a"]
numbers = [10, 20, 10, 30, 20, 10]

print(count_frequencies(words))
print(count_frequencies(numbers))